gitlab-gpu-pipeline.yml

```diff
@@ -0,0 +1,160 @@
+default:
+    tags:
+        - atsccs68-docker-executor
+    image:
+        name: ravilmobile/seissol-base:gcc-8_cuda-10
+        entrypoint: [""]
+
+stages:
+    - pre_build
+    - build 
+    - test
+    - performance_eval
+
+gpu_pre_build:
+    stage: pre_build
+    allow_failure: false
+    variables:
+        GIT_STRATEGY: clone
+
+    before_script:
+        - git branch -vva
+        - echo $commit_author_name
+    script:
+        - echo "HOST arch.:" $HOST
+        - echo "GPU model:" $GPU_MODEL
+        - nvidia-smi
+        - mkdir ./self_usr
+        - git clone https://github.com/uphoffc/ImpalaJIT.git impalajit &&
+          mkdir -p impalajit/build && cd impalajit/build &&
+          cmake .. -DCMAKE_INSTALL_PREFIX=../../self_usr -DCMAKE_CXX_FLAGS="-fPIC" &&
+          make -j $(nproc) install &&
+          cd ../..
+        - git clone --depth 1 --branch yaml-cpp-0.6.2 https://github.com/jbeder/yaml-cpp.git yaml-cpp &&
+          mkdir -p yaml-cpp/build && cd yaml-cpp/build &&
+          cmake .. -DCMAKE_INSTALL_PREFIX=../../self_usr -DCMAKE_CXX_FLAGS="-fPIC" &&
+          make -j $(nproc) install &&
+          cd ../..
```

Comment:
Why don't you just install the module from the package repo?

Reply:
Each job takes a fresh image. It turns out to be the default gitlab-ci policy. I specified a default image for the entire GPU pipeline hoping that gitlab-ci is going to re-use a previously used container but no. In short gitlab-ci doesn't hold an container state between jobs.

`yaml-cpp-0.6.2` is pretty stable package I can move the installation of it to the Dockerfile. What do you think?

Reply:
Yes, that'd be better in my opinion. As long as it it 0.6.x it's going to work -0.7.x might break.

Reply:
Ok. moved yaml-cpp to the docker test image